### PR TITLE
Add Checkstyle Configuration

### PR DESCRIPTION
**Pull request by dzou**
Adds the checkstyle configs for the repo.

### CODE REVIEW

File: src/checkstyle/custom-checks.xml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
+  "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
+
+<module name="Checker">
+  <module name="RegexpHeader">
+    <property name="fileExtensions" value="java"/>
+    <property name="headerFile" value="${checkstyle.header.file}"/>
+  </module>
+  <module name="TreeWalker">
+    <module name="RequireThis">
```
Comment (meltsufin) [inline src/checkstyle/custom-checks.xml:12]: Why do you want to enable `RequireThis`?

File: src/checkstyle/intellij-code-style-scheme.xml [hunk 2]
```diff
@@ -0,0 +1,603 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<!--
+    This file can be imported into IntelliJ to configure editor settings
+    for Google Style conventions.`
+-->
+<code_scheme name="GoogleStyle">
```
Comment (meltsufin) [inline src/checkstyle/intellij-code-style-scheme.xml:7]: What's the original source for this file? Maybe just point people to the source in the readme?

File: src/checkstyle/custom-checks.xml (hunk 1 above)
Comment (dzou) [reply src/checkstyle/custom-checks.xml]: Oops, done removed.

File: src/checkstyle/intellij-code-style-scheme.xml (hunk 2 above)
Comment (dzou) [reply src/checkstyle/intellij-code-style-scheme.xml]: Done - I realized this is linked in the contributing.md so we are good. Removed.

**Comment by dzou** [comment]
> Do we want to keep a copy of intellij settings? Might be better to just refer to https://github.com/google/styleguide/blob/gh-pages/intellij-java-google-style.xml

Good call, removed. I realized that the IDE style configurations are linked in the contributing.md so we are good.

**Pull request merged**